tsne/bh_sne_src/tsne.cpp

```diff
@@ -706,9 +735,28 @@ void run(double* X, int N, int D, double* Y, double perplexity, double theta,
             "%f)!\nLearning embedding...\n",
             (float)(end - start) / CLOCKS_PER_SEC,
             (double)row_P[N] / ((double)N * (double)N));
-  start = clock();
 
-  for (int iter = 0; iter < max_iter; iter++) {
+  auto tsne = std::unique_ptr<TSNEState>(new TSNEState{
```

Comment:
Hm, I guess `make_unique` doesn't work unless you wrote a constructor?  The difference is of course purely theoretical so whatever.  However, the more idiomatic way to do this would be
```suggestion
  unique_ptr<TSNEState> tsne(new TSNEState{
```
or perhaps better yet (to make sure there's no question it's an rvalue)
```suggestion
  return unique_ptr<TSNEState>(new TSNEState{
```

Reply:
Good call.